server/src/main/java/org/opensearch/index/translog/Translog.java

```diff
@@ -1817,6 +1817,13 @@ protected void setMinSeqNoToKeep(long seqNo) {}
 
     protected void onDelete() {}
 
+    /**
+     * Drains ongoing syncs to the underlying store. It returns a releasable which can be closed to resume the syncs back.
+     */
+    protected Releasable drainSyncToStore() {
```

Comment:
Will local translog use this implementation? In that case, a no-op may not be the expected behavior. Until we get the actual implementation added to local translog, can we throw `NotSupportedException`?

Reply:
Local translog will use the above no-op implementation. This code will be executed for non-remote store indexes as well. There are multiple test/ITs failure if we throw exception for local translog case. 

With the no-op implementation, we let the status quo behaviour be there for non-remote store cases. Otherwise we will need to have multiple if else block to run this only for remote indexes.